frontends/ast/simplify.cc

```diff
@@ -1389,10 +1389,8 @@ bool AstNode::simplify(bool const_fold, bool at_zero, bool in_lvalue, int stage,
 
 	if (const_fold && type == AST_CASE)
 	{
-		int width_hint;
-		bool sign_hint;
-		detectSignWidth(width_hint, sign_hint);
 		while (children[0]->simplify(const_fold, at_zero, in_lvalue, stage, width_hint, sign_hint, in_param)) { }
+		detectSignWidth(width_hint, sign_hint);
```

Comment:
Thank you very much for spotting this issue and submitting a PR. Sorry for the delay!

Indeed `width_hint` and `sign_hint` should not have been shadowed here. My mistake!

The ordering of the `detectSignWidth` and the first `simplify` is intentional and swapping it here actually breaks things! Unfortunately, the tests I added didn't appropriately cover this scenario. For example, in the below test case, the spec requires that the case expression is evaluated with the maximum of its own width and that of the case item expressions, meaning that `1'b1 << 1` actually gets simplified to `2'b10` rather than `2'b00`.
```SystemVerilog
module top(
    output logic [5:0] out
);
always_comb
    case (1'b1 << 1)
        2'b10: out = '1;
        default: out = '0;
    endcase
endmodule
```

I have an alternative fix that I'll submit in a PR within the next day or so.